pyart/retrieve/kdp_proc.py

```diff
@@ -21,12 +33,978 @@
+        must be specified in the Py-ART configuration file.
+    band : char, optional
+       Radar frequency band string. Accepted "X", "C", "S" (capital
+       or not). IT is used to compute Rcov and Pcov when not provided    
+    Rcov : 3x3 float array, optional
+        Measurement error covariance matrix
+    Pcov  : 4x4 float array, optional
+        Scaled state transition error covariance matrix
+    prefilter_psidp : bool, optional
+        If set, the psidp measurements will first be filtered with the 
+        filter_psidp method, which can improve the quality of the final Kdp
+    filter_opt : dict, optional
+        The arguments for the prefilter_psidp method, if empty, the defaults
+        arguments of this method will be used
+    parallel : bool, optional
+        Flag to enable parallel computation (one core for every psidp profile)
+        
+    Returns
+    -------
+    kdp_dict : dict
+        Retrieved specific differential phase data and metadata.
+    kdp_std_dict : dict
+        Estimated specific differential phase standard dev. data and metadata.    
+    phidpr_dict,: dict
+        Retrieved differential phase data and metadata.
+
+    References
+    ----------
+    Schneebeli, M., Grazioli, J., and Berne, A.: Improved Estimation
+    of the Specific Differential Phase Shift Using a Compilation of
+    Kalman Filter Ensembles, IEEE T. Geosci. Remote Sens., 52,
+    5137–5149, doi:10.1109/TGRS.2013.2287017, 2014.
+    """
+    
+    # create parallel computing instance 
+    if parallel:
+        import multiprocessing as mp
+        
+        pool = mp.Pool(processes = mp.cpu_count(),maxtasksperchild=1)
+        
+        
+    # parse fill value
+    if fill_value is None:
+        fill_value = get_fillvalue()
+
+    # parse field names
+    if psidp_field is None:
+        psidp_field = get_field_name('differential_phase')
+    if kdp_field is None:
+        kdp_field = get_field_name('specific_differential_phase')
+    if phidp_field is None:
+        phidp_field = get_field_name('differential_phase')
+
+    # parse range resolution, length scale and low-pass filter constraint
+    # weight
+    dr = _parse_range_resolution(radar, check_uniform=True)
+    
+    # parse total differential phase measurements
+    if prefilter_psidp: 
+        psidp_o = filter_psidp(radar,**filter_opt)
+    else:
+        psidp_o = radar.fields[psidp_field]['data']
+    
+    # mask radar gates indicated by the gate filter
+    if gatefilter is not None:
+        psidp_o = np.ma.masked_where(gatefilter.gate_excluded, psidp_o)
+        
+    func = partial(_kdp_kalman_profile,dr = dr, band = band, rcov = rcov, pcov = pcov)
+    
+    all_psidp_prof = list(psidp_o)
+    
+    if parallel:
+        list_est = pool.map(func,all_psidp_prof)
+    else:
+        list_est = map(func,all_psidp_prof)    
+        
+    kdp = np.zeros(psidp_o.shape)*np.nan
+    kdp = np.ma.masked_array(kdp, fill_value = fill_value)
```

Comment:
Is the goal here to create a masked array which is initially completely masked?  If so that is NOT what is being done, and filling a masked array with NaNs is dangerous.  NaNs in masked array start as not masked but can become masked later on if certain checks are triggered.  

A better method to create a fully masked array is:
``` Python
kdp = np.ma.array(np.zeros(phidp_o.shape), mask=True, fill_value=fill_value)
```